### PR TITLE
Implement Walk for gcrane listing

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
Add -r flag to `gcrane {gc,ls}`

### CODE REVIEW

**Comment by jonjohnsonjr** [comment]
@ImJasonH WDYT before I add tests? Generally reasonable?

File: pkg/v1/google/list.go [hunk 1]
```diff
@@ -125,3 +125,58 @@ func List(repo name.Repository, auth authn.Authenticator, t http.RoundTripper) (
...
+		}
+	}
+
+	// We made it!
+	return nil
+}
+
+// Walk recursively descends repositories, calling walkFn.
+// TODO: Do we need a keychain since this can take a while?
+func Walk(root name.Repository, auth authn.Authenticator, t http.RoundTripper, walkFn WalkFunc) error {
```
Comment (ImJasonH) [inline pkg/v1/google/list.go]: I'd love to start standardizing on funcitonal options for new methods especially. In particular `WithAuthFromKeychain` might help solve the `TODO` you have above, and options would let users avoid having to specify `http.DefaultTransport` all the time.

Comment (mattmoor) [reply pkg/v1/google/list.go]: +1

File: pkg/gcrane/list.go [hunk 2]
```diff
@@ -29,50 +27,67 @@ import (
...
+			fmt.Printf("%s:%s\n", repo, tag)
+		}
+		return
+	}
 
 	for _, child := range tags.Children {
 		fmt.Printf("%s/%s\n", repo, child)
 	}
 
+	_ = printImages(repo, tags, err)
```
Comment (ImJasonH) [inline pkg/gcrane/list.go]: Can you add a comment about why this error is unhandled?

File: pkg/v1/google/options.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"log"
+	"net/http"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+)
+
+// WithTransport is a functional option for overriding the default transport
+// on a remote image
+func WithTransport(t http.RoundTripper) ListerOption {
```
Comment (ImJasonH) [inline pkg/v1/google/options.go]: Not a comment on this PR per se, but I think we're going to end up with a bunch of copies of these the way we're going. I'd love to figure out how to reuse them while making it clear which reusable options are usable in which scenarios. I can think of a way of abusing interfaces with private methods, but that's pretty gross. 😷

File: pkg/gcrane/list.go (hunk 2 above)
Comment (jonjohnsonjr) [reply pkg/gcrane/list.go]: Handled it instead.

File: pkg/v1/google/options.go (hunk 3 above)
Comment (jonjohnsonjr) [reply pkg/v1/google/options.go]: Yeah I wasn't sure how to reuse these in a nice way 👎

I figure it's not a ton of code, so some duplication here isn't terrible, but I agree this is less than ideal :)

**Pull request merged**